### PR TITLE
streakcam: MD updater + streak lens

### PR DESCRIPTION
**Pull request by Sabrina85**

### CODE REVIEW

File: src/odemis/gui/model/__init__.py [hunk 1]
```diff
@@ -161,7 +161,12 @@ class MainGUIData(object):
         "e-beam": "ebeam",
         "chamber-light": "chamber_light",
         "overview-light": "overview_light",
-        "pol-analyzer": "pol_analyzer"
+        "pol-analyzer": "pol_analyzer",
+        "streak-cam": "streak_cam",
+        "streak-ccd": "streak_readoutcam",
```
Comment (pieleric) [inline src/odemis/gui/model/__init__.py]: Can we keep it simple and call `streak-ccd` in the code `streak_ccd`? ;-)

File: src/odemis/odemisd/mdupdater.py [hunk 2]
```diff
@@ -359,6 +362,20 @@ def updatePosition(unused, analyzer=analyzer, comp_affected=comp_affected):
 
         return True
 
+    def observeStreakLens(self, streak_lens, comp_affected):
+        """Update the magnification of the streak lens affecting the
+        streak readout camera."""
+
```
Comment (pieleric) [inline src/odemis/odemisd/mdupdater.py]: It works fine now, as long as the only component in the microscope file is the `streak-ccd`. However, to be extra safe, I'd recommend to explicitly check for it, as in `observeLens()`. Something like:
```
     if not comp.role.endswith("ccd"):
            return False
```

File: src/odemis/odemisd/mdupdater.py [hunk 3]
```diff
@@ -359,6 +362,20 @@ def updatePosition(unused, analyzer=analyzer, comp_affected=comp_affected):
...
+        def updateMagnification(unused, streak_lens=streak_lens, comp_affected=comp_affected):
```
Comment (pieleric) [inline src/odemis/odemisd/mdupdater.py]: Actually the `unused` variable is the magnification. So you could rename it `mag`, and avoid reading it again.

Comment (Sabrina85) [reply src/odemis/odemisd/mdupdater.py]: Should we do the same then for the observePol* functions?

Comment (pieleric) [reply src/odemis/odemisd/mdupdater.py]: Humm... indeed, good catch.
Could you have a look at this (in a separate PR)?

**Pull request merged**